Lib/logging/__main__.py

```diff
@@ -0,0 +1,104 @@
+    python scripts, either ones designed with logging in mind or not.
+
+    Solves issue https://github.com/python/cpython/issues/93557
+    -- May Draskovics
+"""
+
+from contextlib import contextmanager   ## contextmanager -> log_exception
+import argparse                         ## argparse.ArgumentParser
+import logging                          ## logging.basicConfig, all other logging functions
+import sys                              ## sys.stdout
+import os                               ## os.path.isfile & os.remove
+
+@contextmanager
+def log_exception(logger:logging.Logger) -> None:
+    """
+        log_exception(logging.Logger) -> None
+            - context manager for running a script with logging
+                and catching exceptions for logging
+
+        :params:
+            logger (logging.Logger) = Current logger for our script
+    """
+    try:
+        yield
+    except Exception as e:
+        logger.exception(e)
+        logger.critical("Script ended on an unhandled exception!")
+
+def log_file(script:str) -> None:
+    """
+        log_file(str) -> None
+            - Runs a python script with logging and enable
+
+        :params:
+            script (str) = The script that we want to log
+    """
+
+
+    logging.info("Logger initialized!")
+
+    """
+        Even with logging we should kill the process
+        if we come into an unhandled exception otherwise
+        we could have some unwanted behavior
+    """
+
+    with log_exception(logging.getLogger()):
+        if script:
+            with open(script, "r") as f:
+                exec(f.read())
+        else:
+            exec(input(">> "))
+        logging.info(f"Script \"{script}\" has ran successfully!")
+        sys.exit(0)
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Python Logging Module")
+    parser.add_argument("-t", "--time_stamp", default=False, action="store_true", help="Include time stamp in output.")
+    parser.add_argument("-p", "--remap_print", default=False, action="store_true", help="Remap the \"print\" function to \"logging.info\".")
+    parser.add_argument("-d", "--double_print", default=False, action="store_true", help="Remap \"print\" to a lambda printing and logging all print statements.")
+    parser.add_argument("-f", "--file", default="", help="Output log to file.")
+    parser.add_argument("-l", "--level", default=0, type=int, help="Log level [1-5] (1=debug, 2=info, 3=warning, 4=error, 5=critical)")
+    parser.add_argument("-s", "--string_format", default="DEFAULT", help="Format for logs.")
+    ## these args aren't used on first run, but are used on subsequent runs
+    parser.add_argument("-r", "--reuse_logger", default=False, action="store_true", help="Reuse the same logger for multiple scripts.")
+    parser.add_argument("-c", "--clear_logfile", default=False, action="store_true", help="Clear the log file before writing to it.")
```

Comment:
We could do both a config file and the args/options. I'm just thinking of how it could be easy to use for anyone. Having a config file could be annoying for some people to add to their projects. But a config file could be added easily.